cmd/kubeadm/app/util/runtime/runtime_unix.go

```diff
@@ -0,0 +1,38 @@
+// +build !windows
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+)
+
+const (
+	dockerSocket     = "/var/run/docker.sock" // The Docker socket is not CRI compatible
+	containerdSocket = "/run/containerd/containerd.sock"
```

Comment:
not sure if we can also use new format for socket URI in this PR: add a prefix `unix://`

Reply:
This was pulled from [detectCRISocketImpl](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/runtime/runtime.go#L195)

Reply:
@chenzhiwei we can't. The unix version of `isExistingSocket` uses the syscalls directly and they take paths, not URIs.